kornia/feature/disk/structs.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+import torch
+import torch.nn.functional as F
+
+from kornia.core import Tensor
+
+
+@dataclass
+class DISKFeatures:
+    r"""A data structure holding DISK keypoints, descriptors and detection scores for an image. Since DISK detects a
+    varying number of keypoints per image, `DISKFeatures` is not batched.
+
+    Args:
+        keypoints: Tensor of shape :math:`(N, 2)`, where :math:`N` is the number of keypoints.
+        descriptors: Tensor of shape :math:`(N, 128)`.
+        detection_scores: Tensor of shape :math:`(N,)` where the detection score can be interpreted as
+        the log-probability of keeping a keypoint after it has been proposed (see the paper,
+        section Method->Feature distribution for details).
+    """
+    keypoints: Tensor
```

Comment:
i didn't get why we had a dataclass for keypoints, and here is just a Tensor. Can't be that struct or https://github.com/kornia/kornia/blob/master/kornia/geometry/keypoints.py#L13?

Reply:
I wasn't aware of `geometry.Keypoints`. Do any other local feature modules use it? It appears that `geometry.Keypoints` is meant as a batchable data container (since it's mostly referenced in data augmentation modules). Why not `disk.Keypoints`? Because I think it's an unnecessary level of indirection, `DISKFeatures` is not cluttered with just 3 members, so merging two of them into a struct of their own won't help usability.

Reply:
@edgarriba do I understand correctly that you're in favor of keeping it as a `Tensor`? I am - `geometry.Keypoints` would be compatible, it would provide some extra functionality (affine transformations) but I don't think this is very useful to the average user and increases the mental burden of searching through documentation.